drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/binary/process/SeleniumServerTestCase.java

```diff
@@ -0,0 +1,136 @@
+                new SeleniumServerBinaryHandler(new DesiredCapabilities()).downloadAndPrepare().toString();
+        capabilities = new DesiredCapabilities();
+        url = new URL("http://localhost:5555/wd/hub/");
+
+        attachLogCapture();
+        setUpStreams();
+    }
+
+    @Test
+    public void should_start_selenium_server_with_serverArgs_debug() throws Exception {
+
+        final String browser = "chrome";
+        final String seleniumServerArgs = "-debug -role node";
+
+        fire(new StartSeleniumServer(seleniumServerBinary, browser, capabilities, url, seleniumServerArgs));
+
+        final String command = parseLogger();
+
+        assertThat(command).contains(seleniumServerArgs);
+        assertThat(outContent.toString()).contains("DEBUG");
+    }
+
+    @Test
+    public void should_start_selenium_server_with_no_serverArgs() throws Exception {
+
+        final String browser = "chrome";
+
+        fire(new StartSeleniumServer(seleniumServerBinary, browser, capabilities, url, null));
+
+        final String command = parseLogger();
+
+        assertThat(command).endsWith(String.valueOf(url.getPort()));
+        assertThat(outContent.toString()).contains("Selenium Server is up and running");
+    }
+
+    @Test
+    public void should_start_selenium_server_as_hub() throws Exception {
+
+        final String browser = "chrome";
+
+        String seleniumServerArgs = "-role hub -browserTimeout 1000";
+
+        fire(new StartSeleniumServer(seleniumServerBinary, browser, capabilities, url, seleniumServerArgs));
+
+        String hubCommand = parseLogger();
+
+        assertThat(hubCommand).contains(seleniumServerArgs);
+        assertThat(outContent.toString()).contains("Selenium Grid hub is up and running");
+    }
+
+    @After
+    public void stopSeleniumServer() {
+        cleanUpStreams();
+        fire(new AfterSuite());
+    }
+
+    private String parseLogger() throws IOException {
+
+        final String capturedLog = getTestCapturedLog();
+        final String expectedLogPart = "Running Selenium server process: java .*$";
+
+        final Pattern pattern = Pattern.compile(expectedLogPart);
+        final Matcher matcher = pattern.matcher(capturedLog);
+        String command;
+
+        if (matcher.find()) ;
+        {
+            command = matcher.group();
+        }
+        return command;
+    }
+
+    private void attachLogCapture() {
+        logCapturingStream = new ByteArrayOutputStream();
+        Handler[] handlers = log.getParent().getHandlers();
+        customLogHandler = new StreamHandler(logCapturingStream, handlers[0].getFormatter());
+        log.addHandler(customLogHandler);
+    }
+
+    private String getTestCapturedLog() throws IOException {
+        customLogHandler.flush();
+        return logCapturingStream.toString();
+    }
+
+    private void setUpStreams() {
+        System.setOut(new PrintStream(outContent));
+    }
+
+    private void cleanUpStreams() {
+        System.setOut(null);
```

Comment:
I think that this is not a clear way to reset values to default - I think, that there should be something like: 
`System.setOut(System.out);`